xcube/webapi/ows/stac/controllers.py

```diff
@@ -302,20 +312,21 @@ def _get_dataset_feature(ctx: DatasetsContext,
         )
 
     t = pyproj.Transformer.from_crs(
-        ml_dataset.grid_mapping.crs,
+        grid_mapping.crs,
         CRS_CRS84,
         always_xy=True
     )
-    bbox = ml_dataset.grid_mapping.xy_bbox
+    bbox = grid_mapping.xy_bbox
     (x1, x2), (y1, y2) = t.transform((bbox[0], bbox[2]), (bbox[1], bbox[3]))
 
     # TODO: Prefer original storage location.
     #       The "s3" operation is default.
     default_storage_url = f"{base_url}/s3/datasets"
 
     if 'time' in dataset:
-        start_time = _format_timestamp(dataset['time'][0].values)
-        end_time = _format_timestamp(dataset['time'][-1].values)
+        time_var = dataset['time']
+        start_time = to_json_value(time_var[0])
+        end_time = to_json_value(time_var[-1])
```

Comment:
Can we guarantee that these will always be datetime64s and not e.g. plain datetimes or pandas Timestamps?

Reply:
No. We can also not guarantee that Pandas timestamps will stay the same. However, xarray time coordinates are `np.datetime64`.

Reply:
OK. If it's enough for our current requirements I guess we can leave it for now and add more handlers to `jsonencoder` if and when the need arises.

Reply:
However, we may enhance `to_json_value()` to also handle Pandas timestamps explicitly.